Prevent diagonal slopes from being built if vehicles lack sprites

mixiate opened the pull request:
This will prevent diagonal slopes from being built if the vehicles don't have the sprites for them. It adds three new track groups, diagSlope, diagSlopeSteepUp and diagSlopeSteepDown, to mirror the non diagonal versions.

This is so that diagonal slopes can't be built without cheats on rides like the go-karts, car ride, ghost train or cycle rides, if and when diagonal track is added, as the vanilla vehicles do not have sprites for them. The steep track groups could be used by the log flume or splash boats, the steep up and down groups are separate so that these tracks can have only down steep track.

Large gentle turns are still able to be built, just without any diagonal track in between them.

tupaschoal commented:
> It's probably best to wait until the go kart pieces PR is up so that this can be tested with them.

Due to this comment, I'll move the PR to draft until that one is up. Can you also link it when it is? Thanks!

mixiate commented:
I've rebased this and double checked that slopes are still buildable on tracks that have them. Added them to the go karts of course.

mixiate commented:
I've rebased it.

ZeeMaji commented:
No issues found with the WW/TT mini suspended vehicles, they work as expected. Will need to do more testing with other rides & vehicles though.

mixiate commented:
It's a fair point. I don't really have an answer for that. An annoying quirk to be sure. Make them cheats only? Wouldn't really want to restrict this even more. Had hoped to get this in before the go karts so this wouldn't happen.

ZeeMaji commented:
It's mildly annoying, but i feel like it's fine. They're still great to have as non-cheat pieces since you can still use them with terrain modifications. Just wanted to note it here incase you or anyone else felt like it was a bigger issue than i felt it was.

ZeeMaji commented:
Also, one option there is available is that the go karts support height could just be increased by one unit as that's all the large slope needs. A similar thing was done with the log flume so you could build a steep drop on it without cheats. Although that would likely need to be a separate PR from this.

mixiate commented:
> This needs a changelog entry and a network bump, since different users would have access to different track pieces.

Ah good catch thank you. I've rebased and added both.

ZeeMaji commented:
Still functions as expected for what it's worth. I'm okay with this being merged as is. Although i would like to see a followup PR that raises the go karts build height as i outlined in my earlier comments.
